src/services/kratos/private.ts

```diff
@@ -0,0 +1,29 @@
+import { getKratosConfig } from "@config"
+import { Configuration, V0alpha2Api, V0alpha2ApiInterface } from "@ory/client"
+
+import { MissingExpiredAtKratosError } from "./errors"
+
+const { publicApi, adminApi } = getKratosConfig()
+
+const KratosSdk: (kratosEndpoint: string) => V0alpha2ApiInterface = (kratosEndpoint) =>
+  new V0alpha2Api(new Configuration({ basePath: kratosEndpoint }))
+
+export const kratosPublic = KratosSdk(publicApi)
+export const kratosAdmin = KratosSdk(adminApi)
+
+export const toDomainSession = (session: KratosSession) => {
+  // is throw ok? this should not happen I (nb) believe but the type say it can
+  // this may probably be a type issue in kratos SDK
+  if (!session.expires_at) throw new MissingExpiredAtKratosError()
```

Comment:
If it's truly something that should never happen throw is okay IMO.
OTOH protecting our code from potentially faulty dependencies is what this layer is about - I'd leave it like this and see if it comes up in prod - then reassess.